protocol/v2/ssv/runner/runner.go

```diff
@@ -68,7 +71,15 @@ func (b *BaseRunner) baseStartNewDuty(runner Runner, duty *spectypes.Duty) error
 	if err := b.canStartNewDuty(); err != nil {
 		return err
 	}
-	b.State = NewRunnerState(b.Share.Quorum, duty)
+
+	// potentially incomplete locking of b.State. runner.Execute(duty) has access to
+	// b.State but currently this does not happen
+	func() {
+		b.mtx.Lock() // writes to b.State
+		defer b.mtx.Unlock()
+		b.State = NewRunnerState(b.Share.Quorum, duty)
+	}()
```

Comment:
Can we simplify to this?

```go
state := NewRunnerState(b.Share.Quorum, duty)
b.mtx.Lock()
b.State = state
b.mtx.Unlock()
```

So that we don't need func + defer, and also this reduces the lock's scope a bit